src/components/ProtectedRoute.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { ReactNode } from 'react';
```

Comment:
If you're able to get the `ProtectedRoutes` functionality from your App component working, you can discard this, as I don't believe it's being used. However, since the related code in your App component isn't working correctly at present, you may want to use this code instead using the techniques from the provided demo.

Reply:
Actually, it looks like this is very similar to the code I provided, but not identical. Just wanted you to be aware of that in case you decide to use this file later.